tensorflow/lite/micro/kernels/one_hot.cc

```diff
@@ -0,0 +1,239 @@
+    const int indices_dims = indices->dims->size;
+    axis = (params->axis == -1) ? indices_dims : params->axis;
+    output_dims = indices_dims + 1;
+    dtype = on_value->type;
+  }
+
+  const TfLiteEvalTensor* indices;
+  const TfLiteEvalTensor* depth;
+  const TfLiteEvalTensor* on_value;
+  const TfLiteEvalTensor* off_value;
+  TfLiteEvalTensor* output;
+
+  int axis;
+  int output_dims;
+  TfLiteType dtype;
+};
+
+// Operation function
+template <typename T, typename TI>
+void OneHotComputeImpl(const OneHotContext& op_context) {
+  int prefix_dim_size = 1;
+  for (int i = 0; i < op_context.axis; ++i) {
+    prefix_dim_size *= op_context.indices->dims->data[i];
+  }
+  if (prefix_dim_size == 0) {
+    return;
+  }
+
+  const RuntimeShape indices_shape =
+      tflite::micro::GetTensorShape(op_context.indices);
+  const int suffix_dim_size = indices_shape.FlatSize() / prefix_dim_size;
+
+  const int depth = *op_context.depth->data.i32;
+
+  const T on_value = *tflite::micro::GetTensorData<T>(op_context.on_value);
+  const T off_value = *tflite::micro::GetTensorData<T>(op_context.off_value);
+
+  T* output_data = tflite::micro::GetTensorData<T>(op_context.output);
+  const TI* indices_data = tflite::micro::GetTensorData<TI>(op_context.indices);
+
+  for (int i = 0; i < prefix_dim_size; ++i) {
+    for (int j = 0; j < depth; ++j) {
+      for (int k = 0; k < suffix_dim_size; ++k, ++output_data) {
+        *output_data =
+            static_cast<int>(indices_data[i * suffix_dim_size + k]) == j
+                ? on_value
+                : off_value;
+      }
+    }
+  }
+}
+
+template <typename T>
+void OneHotCompute(const OneHotContext& op_context) {
+  if (op_context.indices->type == kTfLiteInt64) {
+    OneHotComputeImpl<T, int64_t>(op_context);
+  } else {
+    OneHotComputeImpl<T, int32_t>(op_context);
+  }
+}
+
+TfLiteStatus ResizeOutputTensor(TfLiteContext* context,
+                                const OneHotContext& op_context) {
+  TF_LITE_ENSURE(context, *op_context.depth->data.i32 >= 0);
+
+  // read depth data
+  const int depth_val =
+      *tflite::micro::GetTensorData<int32_t>(op_context.depth);
+  TF_LITE_ENSURE(context, depth_val >= 0);
+
+  // Output Tensor evaluation
+  TF_LITE_ENSURE(context, op_context.output != nullptr);
+
+  TF_LITE_ENSURE(context, op_context.output->dims != nullptr);
+
+  // TFLM assumes that the output tensor’s dims are already allocated
+  const int expected_dims_size = op_context.output_dims;
+  TF_LITE_ENSURE_EQ(context, op_context.output->dims->size, expected_dims_size);
+
+  for (int i = 0; i < expected_dims_size; ++i) {
+    int expected_dim_i;
+    if (i < op_context.axis) {
+      expected_dim_i = op_context.indices->dims->data[i];
+    } else if (i == op_context.axis) {
+      expected_dim_i = depth_val;
+    } else {
+      expected_dim_i = op_context.indices->dims->data[i - 1];
+    }
+
+    // If the size pre-allocated by the TFLM compiler (Offline Memory Planner)
```

Comment:
Change `Offline Memory Planner` to just `Memory Planner` please.  There are several memory planners.